### PR TITLE
Add job to update API doc on gh-pages

### PR DESCRIPTION
**Pull request by KtorZ**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -92,11 +93,12 @@ jobs:
...
-  # - stage: api-doc
-  #   if: branch = develop
-  #   script:
-  #   - travis_wait 42 stack --no-terminal build cardano-sl-wallet-new:exe:cardano-generate-swagger-file --fast --ghc-options=-optl-Wl,-dead_strip_dylibs
-  #   - stack exec -- cardano-generate-swagger-file --target wallet@v1
-  #   - git checkout gh-pages ...
-  #   - ... upload 'swagger.json' to gh-pages
+  # Upload API doc automatically on each build against `develop` to `gh-pages`
+  - stage: documentation
+    if: branch = develop AND type = pull_request # FIXME: type = push
```
Comment (KtorZ) [inline .travis.yml]: NOTE: For testing this through the PR, the `type` has been set to `pull_request`. In practice, we only want to build on `push` (i.e. when something gets merged).

**Review by Anviking — DISMISSED** [review]
lgtm, but aren't we wasting time having this in a separate stage? (wondering not only because of this pr)

**Comment by KtorZ** [comment]
Not much, we only rebuild the source and the swagger executable. That's fairly quick. 
In addition, the idea is to have this only done when things are merged, so no one is really waiting after the CI in such case.

**Comment by KtorZ** [comment]
Could do on nightly too.

**Review by Anviking — APPROVED** [review]

**Pull request merged**